File: articles/agriculture-food-systems-management-life/r/nutrient_use_efficiency.R
# Nutrient use efficiency for synthetic production systems.

systems <- read.csv(file.path("data", "production_systems.csv"), stringsAsFactors = FALSE)

systems$nutrient_use_efficiency <- with(
  systems,
  nutrient_harvested_kg / nutrient_input_kg
)

systems <- systems[order(-systems$nutrient_use_efficiency), ]

print(round(systems[, c("system", "nutrient_input_kg", "nutrient_harvested_kg", "nutrient_use_efficiency")], 4))

File: articles/agriculture-food-systems-management-life/c/nutrient_use_efficiency.c
#include <stdio.h>

// Compact nutrient-use efficiency kernel.

int main(void) {
    double nutrient_input_kg[] = {12000.0, 7600.0, 5200.0};
    double nutrient_harvested_kg[] = {4800.0, 4100.0, 3600.0};
    int n = 3;

    for (int i = 0; i < n; i++) {
        double nue = nutrient_harvested_kg[i] / nutrient_input_kg[i];
        printf("system=%d nutrient_use_efficiency=%.6f\n", i + 1, nue);
    }

    return 0;
}
